.githooks/pre-commit

```diff
@@ -0,0 +1,36 @@
+base_dir='auth0_flutter'
+darwin_dir="$base_dir/darwin"
+staged_files=($(git diff --staged --name-only --diff-filter=ACDRT -- "$darwin_dir"))
```

Comment:
Since we're dealing with symlinks, we only care about files added, copied, renamed, deleted, or files whose type changed:

<img width="681" alt="Screenshot 2023-11-20 at 14 07 11" src="https://github.com/auth0/auth0-flutter/assets/5055789/3f9f8995-98a8-43f6-a550-9f093010af19">